pkg/cloud/aws/actuators/machine/actuator.go

```diff
@@ -69,6 +69,24 @@ type ActuatorParams struct {
 	AwsClientBuilder awsclient.AwsClientBuilderFuncType
 }
 
+// Scan machine tags, and return a deduped tags list
+func removeDuplicatedTags(tags []*ec2.Tag) []*ec2.Tag {
+	m := make(map[string]bool)
+	result := []*ec2.Tag{}
+
+	// look for duplicates
+	for i := range tags {
+		if m[*tags[i].Key] == true {
```

Comment:
You don't need to `== true`.

Reply:
This, I think, can be reduced to:

```
    ...
    keys := make(map[string]bool)
    result := []*ec2.Tag{}
    for _, entry := range tags {
        if _, value := keys[*entry.Key]; !value {
            keys[*entry.Key] = true
            result = append(result, entry)
        }
    }    
    return result
}
```

but untested - I just typed it out!

Please could you also create a unit test for this function as it it generic.